src/main/java/commission/dao/SaleDaoImpl.java

```diff
@@ -6,43 +6,42 @@
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
 
+import javax.sql.DataSource;
 import java.util.List;
 import java.util.Optional;
 
 @Repository
 public class SaleDaoImpl implements SaleDao{
 
-    JdbcTemplate jdbcTemplate;
+    private final JdbcTemplate jdbcTemplate;
+    private final SalesRowMapper rowMapper = new SalesRowMapper();
 
     @Autowired
-    public SaleDaoImpl(JdbcTemplate jdbcTemplate) {
-        this.jdbcTemplate = jdbcTemplate;
+    public SaleDaoImpl(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
     }
 
+
     @Override
     public List<Sale> findAll() {
         var query = "SELECT * FROM sales";
-        return jdbcTemplate.query(query, new SalesRowMapper());
+        return jdbcTemplate.query(query, rowMapper);
```

Comment:
Maybe a bit of an old school comment, but usually class attributes are prefixed with `this.` to indicate it's within that class.

Reply:
Got it!